text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,410 @@
+});
+
+
+new glue.ScalaSparkStreamingJob(this, 'ScalaSparkStreamingJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+});
+
+```
+
+Optionally, developers can override the glueVersion and add extraJars and a
+description:
+
+```
+new glue.PythonSparkStreamingJob(this, 'PythonSparkStreamingJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   description: 'an example Python Streaming job',
+   numberOfWorkers: 20,
+});
+
+
+new glue.ScalaSparkStreamingJob(this, 'ScalaSparkStreamingJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+   description: 'an example Python Streaming job',
+   numberOfWorkers: 20,
+});
+```
+
+3. **Flex Jobs**
+
+The flexible execution class is appropriate for non-urgent jobs such as
+pre-production jobs, testing, and one-time data loads. Flexible job runs
+are supported for jobs using AWS Glue version 3.0 or later and `G.1X` or
+`G.2X` worker types but will default to the latest version of Glue
+(currently Glue 3.0.) Similar to ETL, we’ll enable these features:
+`—enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log`
+
+```
+glue.ScalaSparkFlexEtlJob(this, 'ScalaSparkFlexEtlJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+});
+
+glue.pySparkFlexEtlJob(this, 'pySparkFlexEtlJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+});
+```
+
+Optionally, developers can override the glue version, python version,
+provide extra jars, and a description
+
+```
+glue.ScalaSparkFlexEtlJob(this, 'ScalaSparkFlexEtlJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+   extraJarsS3Url: [glue.Code.fromBucket('bucket-name', 'path-to-extra-python-scripts')],
+   description: 'an example pySpark ETL job',
+   numberOfWorkers: 20,
+});
+
+new glue.pySparkFlexEtlJob(this, 'pySparkFlexEtlJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   description: 'an example Flex job',
+   numberOfWorkers: 20,
+});
+```
+
+### Python Shell Jobs
+
+A Python shell job runs Python scripts as a shell and supports a Python
+version that depends on the AWS Glue version you are using. This can be used
+to schedule and run tasks that don't require an Apache Spark environment.
+
+We’ll default to `PythonVersion.3_9`. Python shell jobs don't support different
+worker typesbut do have they have a MaxDPU feature. Developers can choose
```

Comment:
Can you clarify what you mean here by `Python shell jobs don't support different
worker types`?